ksqldb-parser/src/main/java/io/confluent/ksql/parser/SqlFormatter.java

```diff
@@ -127,9 +127,12 @@ protected Void visitQuery(final Query node, final Integer indent) {
       }
 
       if (!node.isPullQuery()) {
-        if (node.getResultMaterialization().isPresent()) {
+        if (node.getRefinement().isPresent()
+            && node.getRefinement().get().getOutputRefinement().isPresent()) {
           append(indent, "EMIT ");
-          append(indent, node.getResultMaterialization()
+          append(indent, node.getRefinement()
+              .get()
+              .getOutputRefinement()
```

Comment:
this goes back to my comment above that the `OutputRefinement` inside the `RefinementInfo` shouldn't be optional - we expect it to be present. If it's possible that it's not present, then we need to check `isPresent` (which I'm not sure what we'd do if its not)